src/shared/codelens/pythonCodeLensProvider.ts

```diff
@@ -59,75 +72,209 @@ const getLambdaHandlerCandidates = async ({ uri }: { uri: vscode.Uri }): Promise
+                remoteRoot: '/var/task'
+            }
+        ],
+    }
+}
+
+export async function initialize({
     configuration,
-    toolkitOutputChannel,
+    outputChannel: toolkitOutputChannel,
     processInvoker = new DefaultSamCliProcessInvoker(),
     taskInvoker = new DefaultSamCliTaskInvoker()
-}: CodeLensProviderParams): void {
+}: CodeLensProviderParams): Promise<void> {
     const logger = getLogger()
 
-    const runtime = 'python3.7' // TODO: Remove hard coded value
+    const runtime = 'python3.6' // TODO: Remove hard coded value
 
-    const invokeLambda = async (args: LambdaLocalInvokeArguments) => {
+    const invokeLambda = async (args: LambdaLocalInvokeParams) => {
         const samProjectCodeRoot = await getSamProjectDirPathForFile(args.document.uri.fsPath)
-        logger.debug(`Project root: '${samProjectCodeRoot}'`)
+        const baseBuildDir = await makeBuildDir()
+
         let debugPort: number | undefined
 
+        let handlerName: string = args.handlerName
+        let manifestPath: string | undefined
         if (args.isDebug) {
             debugPort = await getDebugPort()
+            const {debugHandlerName} = await makeLambdaDebugFile({
+                handlerName: args.handlerName,
+                debugPort: debugPort,
+                outputDir: samProjectCodeRoot,
+            })
+            handlerName = debugHandlerName
+            manifestPath = await makePythonDebugManifest({
+                samProjectCodeRoot,
+                outputDir: baseBuildDir
+            })
         }
-        // TODO: Figure out Python specific params and create PythonDebugConfiguration if needed
-        const debugConfig: DebugConfiguration = {
-            type: PYTHON_LANGUAGE,
-            request: 'attach',
-            name: 'SamLocalDebug',
-            preLaunchTask: undefined,
-            address: 'localhost',
-            port: debugPort!,
-            localRoot: samProjectCodeRoot,
-            remoteRoot: '/var/task',
-            skipFiles: []
-        }
+        const inputTemplatePath = await makeInputTemplate({
+            baseBuildDir,
+            codeDir: samProjectCodeRoot,
+            documentUri: args.document.uri,
+            handlerName,
+            runtime,
+            workspaceUri: args.workspaceFolder.uri
+        })
+        logger.info(`pythonCodeLensProvider.initialize: ${
+            JSON.stringify({samProjectCodeRoot, inputTemplatePath, handlerName, manifestPath}, undefined, 2)
+        }`)
+
+        const channelLogger = getChannelLogger(toolkitOutputChannel)
+        const codeDir = samProjectCodeRoot
+        const samTemplatePath: string = await executeSamBuild({
+            baseBuildDir,
+            channelLogger,
+            codeDir,
+            inputTemplatePath,
+            manifestPath,
+            samProcessInvoker: processInvoker,
 
-        const localLambdaRunner: LocalLambdaRunner = new LocalLambdaRunner(
+        })
+    
```

Comment:
We should probably make an attempt to remove the file produced by `makeLambdaDebugFile` right here, after executeSamBuild.

What happens if we are unable to delete the file? We should have a file comment heading in that file saying it was autogenerated by the AWS Toolkit for VS Code, so that users can identify its source.

Thinking out loud here -- some alternatives to this (for a future change) could be:
* copy the full samProjectCodeRoot to a temp location within baseBuildDir, write out the xxx_debug.py there, and run sam build against that folder. I don't like this idea in cases where the folder is large.
* don't create xxx_debug.py, run sam build against samProjectCodeRoot, then write xxx_debug.py into the sam build destination and patch the produced template.yaml file. I'm curious what the sam team would think of an idea like this, or what drawbacks this approach might have

Reply:
+1 for the header comment--we should do this regardless of solution (even if we can get this to add directly to the SAM run without a stop in the workspace dir)

Someone voiced that there might be issues with naming clashes, should we add a timestamp or a UUID to the end of the file? Also, could we consider creating the file as a hidden file?

Reply:
Updated to delete `makeLambdaDebugFile` immediately after Lambda invocation. Made name more unique.
✔